modules/matchRoutes.js

```diff
@@ -44,13 +44,13 @@ function getIndexRoute(route, location, callback) {
 }
 
 function assignParams(params, paramNames, paramValues) {
-  return paramNames.reduceRight(function (params, paramName, index) {
+  return paramNames.reduce(function (params, paramName, index) {
```

Comment:
It seems simpler but equivalent to use `reduce` and `push` over `reduceRight` and `unshift`, though.